src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs

```diff
@@ -38,37 +39,45 @@ public StartupLoader(IServiceProvider services, IHostingEnvironment hostingEnv)
             return new StartupMethods(configureMethod.Build(instance), servicesMethod?.Build(instance));
         }
 
-        public Type FindStartupType(string startupAssemblyName, IList<string> diagnosticMessages)
+        public Type FindStartupType(string startupName, IList<string> diagnosticMessages)
         {
             var environmentName = _hostingEnv.EnvironmentName;
-            if (string.IsNullOrEmpty(startupAssemblyName))
+            if (string.IsNullOrEmpty(startupName))
             {
-                throw new ArgumentException("Value cannot be null or empty.", nameof(startupAssemblyName));
+                throw new ArgumentException("Value cannot be null or empty.", nameof(startupName));
             }
 
-            var assembly = Assembly.Load(new AssemblyName(startupAssemblyName));
+            var nameParts = HostingUtilities.SplitTypeName(startupName);
+            var typeName = nameParts.Item1;
+            var assemblyName = nameParts.Item2;
+
+            var assembly = Assembly.Load(new AssemblyName(assemblyName));
             if (assembly == null)
             {
-                throw new InvalidOperationException(String.Format("The assembly '{0}' failed to load.", startupAssemblyName));
+                throw new InvalidOperationException($"The assembly '{assemblyName}' failed to load.");
```

Comment:
BTW for future reference we shouldn't change exception strings to use string interpolation. When these get localized we'll just have to switch back to `string.Format` anyway.